src/agentscope/rag/_store/_store_base.py

```diff
@@ -27,23 +52,194 @@ async def search(
         score_threshold: float | None = None,
         **kwargs: Any,
     ) -> list[Document]:
-        """Retrieve relevant texts for the given queries.
+        """Search for relevant documents based on query embedding.
+
+        All storage types must support vector similarity search as the
+        fundamental retrieval mechanism.
 
         Args:
-            query_embedding (`Embedding`):
-                The embedding of the query text.
-            limit (`int`):
-                The number of relevant documents to retrieve.
-            score_threshold (`float | None`, optional):
-                The threshold of the score to filter the results.
-            **kwargs (`Any`):
-                Other keyword arguments for the vector database search API.
+            query_embedding: The embedding vector of the query
+            limit: Maximum number of documents to retrieve
+            score_threshold: Minimum similarity score threshold (optional)
+            **kwargs: Additional storage-specific search arguments
+
+        Returns:
+            List of relevant documents sorted by relevance
         """
 
     def get_client(self) -> Any:
-        """Get the underlying vector database client, so that developers can
-        access the full functionality of the vector database."""
+        """Get the underlying storage client.
+
+        This allows developers to access the full functionality of the
+        underlying storage system beyond the abstracted interface.
+
+        Returns:
+            The underlying storage client object
+
+        Raises:
+            NotImplementedError: If the storage implementation doesn't
+                                 support direct client access
+        """
+        raise NotImplementedError(
+            f"``get_client`` is not implemented for "
+            f"{self.__class__.__name__}.",
+        )
+
+
+class VDBStoreBase(StoreBase):
```

Comment:
What is the difference between VDBStoreBase and StoreBase? Do we need the StoreBase?